pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -30,6 +30,9 @@ type MeshRootCertificate struct {
 type MeshRootCertificateSpec struct {
 	// Provider specifies the mesh certificate provider
 	Provider ProviderSpec `json:"provider"`
+
+	// TrustDomain is the trust domain to use as a suffix in Common Names for new certificates.
+	TrustDomain string `json:"trustDomain"`
```

Comment:
Have you thought about making the TrustDomain a pointer? That way we could distinguish between the trust domain being set to an empty string vs not being set at all

Reply:
If it is a required field can it be added under `required` in `config_mesh_root_certificate.yaml`?

Reply:
default and required are mutually exclusive. It's explicitly not required, in that it allows a user to omit it, in which case the default value (cluster.local) will be used.

Reply:
from [openapi docs](https://swagger.io/docs/specification/describing-parameters/#common-mistakes-10):


> Common Mistakes
> There are two common mistakes when using the default keyword:
> Using default with required parameters or properties, for example, with path parameters. This does not make sense – if a value is required, the client must always send it, and the default value is never used.
> Using default to specify a sample value. This is not intended use of default and can lead to unexpected behavior in some Swagger tools. Use the example or examples keyword for this purpose instead. See [Adding Examples](https://swagger.io/docs/specification/adding-examples/).
>

Reply:
I missed the default somehow :)